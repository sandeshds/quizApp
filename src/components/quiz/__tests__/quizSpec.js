import React, {Component} from 'react';
import renderer from 'react-test-renderer';
import Quiz from './../index';

describe('Button', ()=>{
    it('should render the entire quiz properly', ()=>{
        let questions = [{
            "text": "This is the first question",
            "options": ["answer1","answer2","answer3","answer4"],
            "answer": 2
        },{
            "text": "This is the second question",
            "options": ["answer 1","answer 2","answer 3", "answer 4"],
            "answer": 1
        }];

        let tree = renderer.create(
            <Quiz 
                allQuestions = {questions}
                fetchQuestions = {() => {}}
                showFetchingQuestionsSpinner = {false}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render fetching questions text properly', ()=>{
        let tree = renderer.create(
            <Quiz 
                allQuestions = {[]}
                fetchQuestions = {() => {}}
                showFetchingQuestionsSpinner = {true}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});