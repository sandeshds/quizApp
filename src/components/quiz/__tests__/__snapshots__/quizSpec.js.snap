// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render fetching questions text properly 1`] = `
<div
  className="quizBlock"
>
  <h2>
     Fetching Questions... 
  </h2>
</div>
`;

exports[`Button should render the entire quiz properly 1`] = `
<div
  className="quizBlock"
>
  <div>
    <div
      className="questionBlock"
    >
      <div>
        <span
          className="questionCounter"
        >
          Javascript Quiz 
          1
           of 
          2
           
        </span>
        <br />
        <span
          className="question"
        >
          This is the first question
        </span>
      </div>
      <div
        className="optionBlock"
      >
        <div
          className="option"
        >
          <div>
            A
             - 
            answer1
          </div>
        </div>
        <div
          className="option"
        >
          <div>
            B
             - 
            answer2
          </div>
        </div>
        <div
          className="option"
        >
          <div>
            C
             - 
            answer3
          </div>
        </div>
        <div
          className="option"
        >
          <div>
            D
             - 
            answer4
          </div>
        </div>
      </div>
    </div>
    <div
      className="optionsBlock"
    >
      <input
        className="button"
        onClick={[Function]}
        type="button"
        value="A"
      />
      <input
        className="button"
        onClick={[Function]}
        type="button"
        value="B"
      />
      <input
        className="button"
        onClick={[Function]}
        type="button"
        value="C"
      />
      <input
        className="button"
        onClick={[Function]}
        type="button"
        value="D"
      />
    </div>
  </div>
</div>
`;
